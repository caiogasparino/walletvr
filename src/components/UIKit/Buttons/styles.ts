import {TouchableOpacity, Text} from 'react-native';
import styled from 'styled-components/native';
import {IButtonProps} from './IButtons';

export const ButtonContainer = styled(TouchableOpacity)<IButtonProps>`
  background-color: ${props => props.backgroundColor || '#007AFF'};
  justify-content: center;
  border-radius: ${props => props.borderRadius || 5}px;
  padding-vertical: ${props => props.paddingVertical || 10}px;
  padding-horizontal: ${props => props.paddingHorizontal || 20}px;
`;

export const ButtonText = styled(Text)<{color?: string}>`
  color: ${props => props.color || '#fff'};
  font-size: 16px;
  text-align: center;
`;
