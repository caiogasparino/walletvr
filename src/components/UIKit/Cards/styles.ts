import styled from 'styled-components/native';
import {ICardProps} from './ICardProps';
import {Colors} from '@UIKit';

export const CardContainer = styled.View<ICardProps>`
  background-color: ${props =>
    props.cardType === 'Black Card' ? Colors.primaryBlack : Colors.cardBg};
  border-radius: 16px;
  padding: 10px;
  gap: 10px;
  width: 300px;
  height: 180px;
  margin: 16px;
  opacity: ${props => props.opacity || 1};
  box-shadow: 0px 0px 12px rgba(0, 0, 0, 0.25);
`;

export const Contain = styled.View`
  display: flex;
  width: 100%;
  padding-top: 10px;
  align-items: flex-start;
`;
export const TitleContain = styled.View`
  display: flex;
  width: 100%;
  padding: 10px;

  align-items: flex-start;
`;
export const TextContain = styled.View`
  display: flex;
  width: 100%;
  padding: 10px;
  gap: 1px;
  align-items: flex-start;
`;
