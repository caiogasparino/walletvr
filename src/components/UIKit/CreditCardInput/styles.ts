import styled from 'styled-components/native';
import {Colors} from '@UIKit';

export const Container = styled.View`
  flex-direction: row;
  align-items: center;
  background: ${Colors.primaryWhite};
  height: 45px;
  padding-horizontal: 12px;
  border-radius: 5px;
`;

export const TypographyContain = styled.View`
  display: flex;
  width: 100%;
  align-items: flex-start;
`;

export const CamContain = styled.TouchableOpacity`
  display: flex;
  align-items: flex-start;
`;

export const InputCard = styled.TextInput`
  flex: 1;
  margin-left: 10px;
  font-size: 16px;
  color: ${Colors.primaryBlack};
  font-family: 'PTSansCaption-Regular';
`;
