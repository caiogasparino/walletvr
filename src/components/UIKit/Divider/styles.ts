import styled from 'styled-components/native';
import {CustomDividerProps} from './IDivider';

export const DividerContainer = styled.View<CustomDividerProps>`
  padding-bottom: ${({paddingBottom}) => `${paddingBottom || 0}px`};
  border-color: ${({borderColor}) => borderColor || 'transparent'};
  border-bottom-width: ${({borderBottomWidth}) =>
    `${borderBottomWidth || 0}px`};
`;

export const DividerContainerShadow = styled.View<CustomDividerProps>`
  padding-bottom: ${({paddingBottom}) => `${paddingBottom || 0}px`};
  border-color: ${({borderColor}) => borderColor || 'transparent'};
  border-bottom-width: ${({borderBottomWidth}) =>
    `${borderBottomWidth || 0}px`};
  elevation: ${({elevation}) => `${elevation || 0}`};
  shadow-color: ${({shadowColor}) => shadowColor || 'transparent'};
  shadow-opacity: ${({shadowOpacity}) => shadowOpacity || 0};
  shadow-radius: ${({shadowRadius}) => `${shadowRadius || 0}px`};
  shadow-offset: ${({shadowOffset}) =>
    shadowOffset
      ? `${shadowOffset.width || 0}px ${shadowOffset.height || 0}px`
      : '0px 0px'};
`;
