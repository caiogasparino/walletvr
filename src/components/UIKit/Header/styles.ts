import styled from 'styled-components/native';

export const Container = styled.View`
  position: absolute;
  width: 65%;
  top: 60px;
  left: 0px;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

export const ButtonArrow = styled.TouchableOpacity``;
