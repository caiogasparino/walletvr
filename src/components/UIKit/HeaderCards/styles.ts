import styled from 'styled-components/native';
import {Colors} from '@UIKit';

const commonHeaderStyles = `
  background-color: ${Colors.primaryWhite};
  z-index: 1;
  width: 100%;
`;

export const Container = styled.View`
  ${commonHeaderStyles}
  height: 110px;
  position: absolute;
  top: 0;
  padding-horizontal: 8%;
  padding-top: 10%;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

export const ContainerHeaderTitle = styled.View`
  ${commonHeaderStyles}
  height: 70px;
  top: 110px;
  position: absolute;
  justify-content: center;
  align-items: center;
  border-bottom-left-radius: 50px;
  border-bottom-right-radius: 50px;
  shadow-color: ${Colors.shadowColor};
  shadow-opacity: 0.8;
  shadow-radius: 10px;
  shadow-offset: 0px 4px;
  elevation: 5;
`;

export const ButtonArrow = styled.TouchableOpacity``;
