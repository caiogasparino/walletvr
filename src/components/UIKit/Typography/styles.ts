// styles.ts

import styled from 'styled-components/native';
import {ITypography} from './ITypography';

export const TextCustom = styled.Text<ITypography>`
  text-align: ${({align}) => align || 'left'};
  font-size: ${({size}) => `${size || 16}px`};
  color: ${({color}) => color || 'black'};
  text-decoration: ${({decoration}) => decoration || 'none'};
  line-height: ${({lineHeight}) => `${lineHeight || 22}px`};
  opacity: ${({opacity}) => opacity || 1};
`;
