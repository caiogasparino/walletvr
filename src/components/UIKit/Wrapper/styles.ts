import styled from 'styled-components/native';
import LinearGradient from 'react-native-linear-gradient';
import {Normalize, Colors} from '../';
import {IWrapper} from './IWrapper';

export const Gradient: any = styled(LinearGradient).attrs({
  colors: [Colors.primary, Colors.primary],
  start: {x: 0, y: 0},
  end: {x: 0, y: 1},
})<IWrapper>`
  flex: 1;
  padding-top: ${props => Normalize(props.pt || 0)}px;
  padding-bottom: ${props => Normalize(props.pb || 0)}px;
`;

export const White = styled.View<IWrapper>`
  background: ${Colors.primaryBg};
  flex: 1;
  padding-top: ${props => Normalize(props.pt || 0)}px;
  padding-bottom: ${props => Normalize(props.pb || 0)}px;
`;

export const Transparent = styled.View<IWrapper>`
  background: transparent;
  flex: 1;
  padding-top: ${props => Normalize(props.pt || 0)}px;
  padding-bottom: ${props => Normalize(props.pb || 0)}px;
`;
