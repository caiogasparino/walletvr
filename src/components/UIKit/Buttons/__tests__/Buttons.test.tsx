// Button.test.tsx

import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import '@testing-library/jest-native/extend-expect';
import renderer from 'react-test-renderer';
import {Button} from '@UIKit';
import {ButtonContainer} from '../styles';

describe('Button Component', () => {
  test('renders with the correct label', () => {
    const {getByText} = render(<Button label="Click me" />);
    expect(getByText('Click me')).toBeDefined();

    const tree = renderer.create(<Button label="Click me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('calls onPress prop when clicked', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(
      <Button label="Click me" onPress={onPressMock} />,
    );

    fireEvent.press(getByText('Click me'));

    expect(onPressMock).toHaveBeenCalledTimes(1);

    const tree = renderer
      .create(<Button label="Click me" onPress={onPressMock} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders with disabled attribute when disabled is true', () => {
    const {getByTestId} = render(
      <ButtonContainer testID="myButton">
        <Button label="Click me" disabled={true} />
      </ButtonContainer>,
    );

    const button = getByTestId('myButton');
    const innerButton = button.children[0];
    expect(innerButton).toHaveProp('disabled', true);

    const tree = renderer
      .create(<Button label="Click me" testID="myButton" disabled={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
