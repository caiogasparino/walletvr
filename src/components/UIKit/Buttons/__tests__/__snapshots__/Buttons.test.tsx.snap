// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component calls onPress prop when clicked 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#007AFF",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "height": undefined,
      "justifyContent": "center",
      "marginBottom": 0,
      "marginTop": 0,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
      "width": undefined,
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": undefined,
          "fontFamily": "PTSansCaption-Regular",
          "fontSize": undefined,
          "fontWeight": undefined,
          "lineHeight": 22,
        }
      }
    >
      Click me
    </Text>
  </Text>
</View>
`;

exports[`Button Component renders with disabled attribute when disabled is true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#007AFF",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "height": undefined,
      "justifyContent": "center",
      "marginBottom": 0,
      "marginTop": 0,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
      "width": undefined,
    }
  }
  testID="myButton"
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": undefined,
          "fontFamily": "PTSansCaption-Regular",
          "fontSize": undefined,
          "fontWeight": undefined,
          "lineHeight": 22,
        }
      }
    >
      Click me
    </Text>
  </Text>
</View>
`;

exports[`Button Component renders with the correct label 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#007AFF",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "height": undefined,
      "justifyContent": "center",
      "marginBottom": 0,
      "marginTop": 0,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
      "width": undefined,
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": undefined,
          "fontFamily": "PTSansCaption-Regular",
          "fontSize": undefined,
          "fontWeight": undefined,
          "lineHeight": 22,
        }
      }
    >
      Click me
    </Text>
  </Text>
</View>
`;
